### PR TITLE
ci(sdk): build wheels on musllinux

### PR DESCRIPTION
**Pull request by dmitryduev**
Description
-----------

- Fixes WB-18448

Status: IT DOESN'T WORK
image

Problems:
- Compiling nvml with cgo+musl fails, and we've been so far unable to force any linker (lld or gold) to not worry about unresolved symbols (that's how nvml is written...). Even tried zig, but it also has issues with musl.

- Tried skipping nvidia monitoring completely, it builds successfully, but the repair wheel step fails:

```
Repairing wheel...
      + sh -c 'auditwheel repair --lib-sdir . -w /tmp/cibuildwheel/repaired_wheel /tmp/cibuildwheel/built_wheel/wandb-0.17.0rc4-py3-none-linux_x86_64.whl'
  INFO:auditwheel.main_repair:Repairing wandb-0.17.0rc4-py3-none-linux_x86_64.whl
  Traceback (most recent call last):
    File "/usr/local/bin/auditwheel", line 8, in <module>
      sys.exit(main())
    File "/opt/_internal/pipx/venvs/auditwheel/lib/python3.10/site-packages/auditwheel/main.py", line 54, in main
      rval = args.func(args, p)
    File "/opt/_internal/pipx/venvs/auditwheel/lib/python3.10/site-packages/auditwheel/main_repair.py", line 125, in execute
      wheel_abi = analyze_wheel_abi(wheel_policy, wheel_file, exclude)
    File "/opt/_internal/pipx/venvs/auditwheel/lib/python3.10/site-packages/auditwheel/wheel_abi.py", line 246, in analyze_wheel_abi
      ) = get_wheel_elfdata(wheel_policy, wheel_fn, exclude)
    File "/opt/_internal/pipx/venvs/auditwheel/lib/python3.10/site-packages/auditwheel/wheel_abi.py", line 87, in get_wheel_elfdata
      for key, value in elf_find_versioned_symbols(elf):
    File "/opt/_internal/pipx/venvs/auditwheel/lib/python3.10/site-packages/auditwheel/elfutils.py", line 50, in elf_find_versioned_symbols
      for verneed, verneed_iter in section.iter_versions():
  AttributeError: 'Section' object has no attribute 'iter_versions'
                                                              ✕ 3.04s
Error: Command ['sh', '-c', 'auditwheel repair --lib-sdir . -w /tmp/cibuildwheel/repaired_wheel /tmp/cibuildwheel/built_wheel/wandb-0.17.0rc4-py3-none-linux_x86_64.whl'] failed with code 1. 
```

- [ ] I updated CHANGELOG.md, or it's not applicable

Testing
-------
How was this PR tested?

### CODE REVIEW

File: .github/workflows/release-sdk.yml [hunk 1]
```diff
@@ -115,7 +115,7 @@ jobs:
           # Not sure why we skip PyPy and musllinux builds.
           #
           # See https://cibuildwheel.readthedocs.io/en/stable/options/#build-skip
-          CIBW_SKIP: cp36-* pp* *musllinux*
+          CIBW_SKIP: cp36-* pp*
```
Comment (timoffex) [inline .github/workflows/release-sdk.yml:118]: Can probably also remove "cp36-" and the comment about it. We set the Python min version in `pyproject.toml`.

**Review by timoffex — APPROVED** [review]
LGTM if it all works!

File: .github/workflows/release-sdk.yml (hunk 1 above)
Comment (dmitryduev) [reply .github/workflows/release-sdk.yml]: yea, that works!

**Comment by dmitryduev** [comment]
It does now!
image

**Pull request closed without merging**